### ISSUE TITLE
Add Price field/filter

### ISSUE THREAD

**Issue by eklam** [AdventureLookup/AdventureLookup#225]
A Price field Min-Max would be greatly appreciated.

**Comment by marcellarius** [comment]
I think Matt has previously weighed in on this; the problem is that prices can change and it's something that would need to be manually kept up to date.

But some way of tagging free adventures would probably be quite useful, and information that is less likely to change.

**Comment by hobk1ard** [comment]
I agree, just flag free adventures. People should pretty readily report if this changes.

**Comment by burnhamrobertp** [comment]
"Free" can change as well, though, and by the retailer/seller. It could also be free somewhere but not conveniently available for some reason.

**Comment by hobk1ard** [comment]
Could you just specify it must be free at the provided link? If that changes, give an easy way for people to flag that it has changed. Maybe something outside the current CR process?

I just see an option like this being heavily used in searches, so it is worth discussing/trying.

**Comment by JohnnyFlash** [comment]
Also, is 99 cents so expensive that it should be in the category of not free?

The idea here is we don’t want to punish adventures for not being free. Let the users find the best adventure based on their search and decide if they want to get it.

________________________________

**Comment by hobk1ard** [comment]
That is a fair point, but i still think tracking a boolean like free has it value and would be attractive to a lot of people. However, this isn't my vision/project. I do love what has already been done.

Thanks for replying. If I can find the time I will try to help contribute.

**Comment by alexrawlins** [comment]
Perhaps something a bit more general than specific prices like Yelp has the $, $$, and $$$ categories. The difficulty with that is that prices vary by outlet (PDF vs Amazon vs Local Store vs Used Copy). So unless you tie the price to the link and allow multiples I am not sure how you make that possible, and that is a lot of work.

It might be almost easier to list a general availability and rely on the searcher to follow the link. Something of a discrete list like: PDF available, PDF and Hard Copy in Print, PDF and Hard Copy out of print, Out of Print. So that way I can search by something that has a PDF known to be purchasable, or something where the hard copy is in print.

**Comment by JohnnyFlash** [comment]
Of all the suggestions to come out of this thread,  I like your availability list best as it seems like real information that is useful to the searchers.

-Jerry

**Comment by marcellarius** [comment]
Free is fundamentally different from cheap. If I'm hunting through adventures and just want some ideas then I can look at 10 free adventures, but I wouldn't pay $1 each for them. 

Plus there are a lot of people who play D&D but can't make these purchases — either due to very low income, or just not having a credit card (too young, etc.)

**Comment by burnhamrobertp** [comment]
There is a strong argument as to the brittleness of storing price info in AL which would require additional technical work, increase system complexity, and would require constant human attention that far exceeds its value. This same information can be obtained just by a few extra clicks or a google search for the product(s) in question, so its not as if there isn't a reasonable-effort option, either. Lastly, Matt himself has vocalized an opinion mirroring the thoughts I've written above, and in the absence of an argument strong enough to change either Matt's opinion or the circumstances of the difficulties associated with storing price, we would treat this request as `won't implement`.

I'll close this ticket, but I appreciate everyone's involvement in the discussion, and certainly if the other contributors, Matt, or Jerry disagree with my summary (either now or in the future), they can reopen this ticket as necessary.

**Comment by tholum** [comment]
@burnhamrobertp If a patch was submitted for a system that allowed user driven price. Would it be considered? Not being able to see the price, at least for me makes the system very unusable.

**Comment by JohnnyFlash** [comment]
This is my opinion on this topic and I feel that it is backed by Matt as well.

This is an open source project. If you wanted to add that functionality to the code base as long as it fit the code base and passed code review it should be allowed.

However, one requirement would be for the option to disable that functionality in the preferences, because AdventureLookup.com <http://adventurelookup.com/> is not planning on displaying prices for the foreseeable future.

We’re not saying your desire to have this feature is wrong.  We are saying we don’t want this feature on our website.

-Jerry

**Comment by burnhamrobertp** [comment]
Along these lines, I would put forth the notion that functionality that's known to be disabled may not be good for the code. Every line increases the costs of maintenance and increases the risk of introducing regressions and so on. This is something that you can really only know once you've suffered from large codebase projects, so I'd want to make sure this has been considered before we take on an official stance for functionality that seems like it's contentious and may not ever be enabled.

**Comment by tholum** [comment]
@JohnnyFlash Is there any way to do a sync of the data from adventurelookup.com to a different site? Because without the adventures in the database the mod would not be anywhere near as useful.

**Comment by burnhamrobertp** [comment]
Not presently, that would require a data dump with us manually purging out sensitive data (users, most obviously). And then you would load that mysql dump in a destination database. The process of that is well documented, its just a factor of us pulling the backup and excluding the sensitive tables' data. In fact, because of the necessary relationships in other tables, we'd actually have to keep the users table but de-identify it and remove all passwords. We'd have to do that in a separate copy so as to not effect current production.

In short, it'd be a lot of work for whomever.
